sdk/appcenter-identity/src/main/java/com/microsoft/appcenter/identity/Identity.java

```diff
@@ -482,9 +470,11 @@ public void run() {
 
     @UiThread
     private synchronized void signInInteractively() {
-        if (mActivity != null) {
+        if (!NetworkStateHelper.getSharedInstance(mContext).isNetworkConnected()) {
```

Comment:
This check is done too late, silent sign in would fail or be delayed without network so need to be applied earlier too.

Reply:
Should be both checks at same place.

Reply:
iOS does it correctly: https://github.com/Microsoft/AppCenter-SDK-Apple/pull/1348/files#diff-eb504461a7342631bb7a7693117de49cR188